test/e2e/storage/csi_mock/csi_volume_expansion.go

```diff
@@ -201,6 +201,85 @@ var _ = utils.SIGDescribe("CSI Mock volume expansion", func() {
 			})
 		}
 	})
+	ginkgo.Context("CSI online volume expansion with secret[Feature:CSINodeExpandSecret", func() {
+		tests := []struct {
+			name          string
+			disableAttach bool
+		}{
+			{
+				name: "should expand volume without restarting pod if attach=on, nodeExpansion=on, csiNodeExpandSecret=on",
+			},
+		}
+		for _, t := range tests {
+			test := t
+			ginkgo.It(test.name, func(ctx context.Context) {
+				var err error
+				params := testParameters{enableResizing: true, enableNodeExpansion: true, enableCSINodeExpandSecret: true}
+				if test.disableAttach {
+					params.disableAttach = true
+					params.registerDriver = true
+				}
+
+				m.init(ctx, params)
+				ginkgo.DeferCleanup(m.cleanup)
+
+				manifests := []string{
+					"test/e2e/testing-manifests/storage-csi/node-expand-volume-secret/secret.yaml",
+				}
+				err = utils.CreateFromManifests(ctx, f, m.config.Framework.Namespace, nil, manifests...)
+				if err != nil {
+					framework.Failf("deploying secret: %v", err)
+				}
+				sc, pvc, pod := m.createPod(ctx, pvcReference)
+				gomega.Expect(pod).NotTo(gomega.BeNil(), "while creating pod for resizing")
+
+				if !*sc.AllowVolumeExpansion {
+					framework.Fail("failed creating sc with allowed expansion")
+				}
+				if sc.Parameters == nil {
+					framework.Fail("failed creating sc with secret")
+				}
+				if _, ok := sc.Parameters[nodeExpandSecretKey]; !ok {
+					framework.Failf("creating sc without %s", nodeExpandSecretKey)
+				}
+				if _, ok := sc.Parameters[nodeExpandSecretNamespaceKey]; !ok {
+					framework.Failf("creating sc without %s", nodeExpandSecretNamespaceKey)
+				}
+				pv, err := m.cs.CoreV1().PersistentVolumes().Get(ctx, pvc.Spec.VolumeName, metav1.GetOptions{})
+				if err != nil {
+					framework.Failf("failed to get pv %s, %v", pvc.Spec.VolumeName, err)
+				}
+				if pv.Spec.CSI == nil || pv.Spec.CSI.NodeExpandSecretRef == nil {
+					framework.Fail("creating pv without 'NodeExpandSecretRef'")
+				}
+				err = e2epod.WaitForPodNameRunningInNamespace(ctx, m.cs, pod.Name, pod.Namespace)
+				framework.ExpectNoError(err, "Failed to start pod1: %v", err)
+
+				ginkgo.By("Expanding current pvc")
+				newSize := resource.MustParse("6Gi")
+				newPVC, err := testsuites.ExpandPVCSize(ctx, pvc, newSize, m.cs)
```

Comment:
Do we have a way to check that the secret was passed through to the CSI driver?

Reply:
we (@zhucan )  were discussing that and if possible may be just log it . more or less it can also be a followup/refresh later.

Reply:
@xing-yang do you know how the snapshot secret tests handle this?

Reply:
For snapshot secrets testing, It was done using a PreHook:
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/csi_mock/csi_snapshot.go#L210

Reply:
Check the secret in the snapshotcontent, not from the csi logs? @xing-yang

Reply:
It's done using a hook, similar to this one: https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/csi_mock/csi_node_stage_error_cases.go#L37